view/frontend/web/js/view/payment/method-renderer/omise-offsite-alipay-method.js

```diff
@@ -44,64 +44,121 @@ define(
              *
              * @return {boolean}
              */
-            isActive: function() {
+            isActive: function () {
                 return true;
             },
 
+            /**
+             * Get Omise public key
+             *
+             * @return {string}
+             */
+            getPublicKey: function () {
+                return window.checkoutConfig.payment.omise_cc.publicKey;
+            },
+
+            /**
+             * Get order total amount
+             *
+             * @return {string}
+             */
+            getOrderAmount: function () {
+                return window.checkoutConfig.quoteData.grand_total * 100;
+            },
+
+            /**
+             * Get order currency
+             *
+             * @return {string}
+             */
+            getOrderCurrency: function () {
+                return window.checkoutConfig.quoteData.quote_currency_code;
+            },
+
+            /**
+             * Initiate observable fields
+             *
+             * @return this
+             */
+            initObservable: function () {
+                this._super()
+                    .observe([
+                        'omiseSourceToken'
+                    ]);
+
+                return this;
+            },
+
+            /**
+             * Get a checkout form data
+             *
+             * @return {Object}
+             */
+            getData: function () {
+                return {
+                    'method': this.item.method,
+                    'additional_data': {
+                        'omise_source_token': this.omiseSourceToken()
+                    }
+                };
+            },
+
             /**
              * Hook the placeOrder function.
              * Original source: placeOrder(data, event); @ module-checkout/view/frontend/web/js/view/payment/default.js
              *
              * @return {boolean}
              */
-            placeOrder: function(data, event) {
+            placeOrder: function (data, event) {
                 var self = this;
 
                 if (event) {
                     event.preventDefault();
```

Comment:
I prefer one-liners for `if`s like this, but not sure what our style guide says

Reply:
that is how format JS Visual Studio code, I prefere it much more, when we have this type of JS function declaration. Also in arrow functions I do the same, It looks much more clear.

Reply:
space is better in my opinion, the same you do when you use arrow functions with async
```
someFunc: async (req, res) => {
}
or
someFunc: function (req, res) {

}
```

Reply:
?? I was talking about the `if` statement, but appear to have highlighted the wrong line

Reply:
well as you can see that are not my changes.. I did not touch that part...
I might have opportunity to look how to improve entire code later.. but not having time to do it in current sprint.